### ISSUE TITLE
Support a base-url for path-based routing

### ISSUE THREAD

**Issue by docktermj** [Senzing/senzing-api-server#214]
**Is your feature request related to a problem? Please describe.**
Currently when running a `senzing/senzing-api-server` docker container, the application is hard-coded to the root of the container.  In other words, it is accessed via URL like `http://localhost:8250`.   Although this works in many situations, it does not work well in docker formations that use path-based routing.

**Describe the solution you'd like**
Have `senzing/senzing-api-server` support the `SENZING_BASE_URL_SENZING_API_SERVER` environment variable and `-base-url` command line option.

So without the environment variable or command  line option, a CURL call would look like:

```console
curl -X GET http://localhost:8250/heartbeat
```

But if `SENZING_BASE_URL_SENZING_API_SERVER=/api/`, then the CURL call would look like

```console
curl -X GET http://localhost:8250/api/heartbeat
```

**Additional context**
For more information, see https://github.com/Senzing/docker-xterm/issues/16

**Comment by docktermj** [comment]
Related issue: https://github.com/Senzing/senzing-rest-api-specification/issues/60

**Comment by docktermj** [comment]
@barrycaceres, you should be able to test your "path-based routing" code with docker-compose by following the instructions at:
https://github.com/Senzing/docker-compose-demo/tree/master/docs/docker-compose-rabbitmq-postgresql-advanced

Using:
```
export SENZING_DOCKER_COMPOSE_FILE=resources/postgresql/docker-compose-rabbitmq-postgresql-path-based-routing.yaml
```

Then try the following URLs:

http://localhost:9180/swagger - Done by Swagger folks
http://localhost:8254/xterm 
http://localhost:8250/api 
http://localhost:8251/app

**Comment by barrycaceres** [comment]
This is addressed in version 2.5.0 of the Senzing REST API Server.

The command line option(s) are `--url-base-path` or alternatively `-urlBasePath` to match the old command-line parameter format.

The environment variable is `SENZING_API_SERVER_URL_BASE_PATH`.

The value of the base path can include or omit leading or trailing forward slashes.  That is to say, the following are equivalent:

- `/api`
- `/api/`
- `api/`
- `api`
